memecity.game/Game/States/GameState.cpp

```diff
@@ -1,19 +1,49 @@
 #include "GameState.h"
-#include "..\..\Assets.h"
+#include "../../Assets.h"
+#include "../Systems/FightingSystem.h"
+
+GameState::GameState(memecity::engine::state::StateManager& sm, GameManager::GameContext& gc,
+	memecity::engine::ecs::EntityManager em) : _context(&gc), entity_manager(std::move(em)), State(sm), _hud(_context->get_multimedia_manager(), _context->get_multimedia_manager().get_texture("Sprites/big_black.bmp", 0, 0, _context->get_multimedia_manager().get_screen_width(), 100), 0, 0)
+{
+
+	FightingSystem* system = dynamic_cast<FightingSystem*>(entity_manager.get_system_of_type<FightingSystem>());
```

Comment:
deze cast kan je beter in in de `get` doen. Ik weet dat ik die get nog ga aanpassen, maar als jij dit al doet, veranderd er alleen intern dingen, en hoef je extern niks te veranderen.